modules/api/src/main/java/org/apache/ignite/tx/Transaction.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.tx;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * The transaction.
+ */
+public interface Transaction {
```

Comment:
Would you mind making it AutoClosable? 
This will be really helpful in case of synchronous usage.
`
try (Transaction tx = transactions.beginTx()) {
  ....
}
`

Reply:
No, I wouldn't. Explained why on dev-list discussion.